### PR TITLE
[improvement](Jsonb) optimization Jsonb path parse

### PR DESCRIPTION
**Pull request by LemonLiTree**
The previous logic was to read jsonbvalue while parsing the json path. For complex json paths, there will be a lot of repeated parsing work. The optimization idea is to separate the analysis and value of jsonpath

### CODE REVIEW

**Comment by LemonLiTree** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 49.63 seconds
 stream load tsv:          455 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          70.1 seconds inserted 10000000 Rows, about 142K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705071911_clickbench_pr_172557.html

**Comment by eldenmoon** [comment]
could you explain more details of optimization logic and  provide some performance diff?

**Review by eldenmoon — COMMENTED** [review]
LGTM

File: be/src/vec/functions/function_jsonb.cpp [hunk 1]
```diff
@@ -526,8 +523,13 @@ struct JsonbExtractStringImpl {
             int r_size = roffsets[i] - roffsets[i - 1];
             const char* r_raw = reinterpret_cast<const char*>(&rdata[roffsets[i - 1]]);
 
+            JsonbPath path;
+            if (!path.seek(r_raw, r_size)) {
+                is_invalid_json_path = true;
```
Comment (amorynan) [inline be/src/vec/functions/function_jsonb.cpp:528]: if is_invalid_json_path = true , whether we do not need to call inner_loop_impl func ,and just push_null_string & return ?

**Comment by LemonLiTree** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.64 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17161674573 Bytes

**Review by amorynan — APPROVED** [review]
LGTM

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**